pyinteraph/centrality_analysis.py

```diff
@@ -0,0 +1,604 @@
+        except FileNotFoundError:
+            raise FileNotFoundError(f"PDB not found: {pdb}")
+        except:
+            raise Exception(f"Could not parse pdb file: {pdb}")
+        # generate identifiers for the nodes of the graph
+        identifiers = [f"{r.segment.segid}{r.resnum}" for r in u.residues]
+    # if the user did not provide a reference structure
+    else:
+        # generate automatic identifiers going from 1 to the
+        # total number of residues considered
+        identifiers = [f"_{i}" for i in range(1, adj_matrix.shape[0]+1)]
+
+    # generate a graph from the data loaded
+    G = nx.Graph(adj_matrix)
+    # set the names of the graph nodes (in place)
+    node_names = dict(zip(range(adj_matrix.shape[0]), identifiers))
+    nx.relabel_nodes(G, mapping = node_names, copy = False)
+    # return the identifiers and the graph
+    return identifiers, G
+
+def convert_input_to_list(user_input, identifiers):
+    """Take in a string (e.g. A12:A22,A13... if a PDB file is supplied)
+    and a list of names of all the residues (graph nodes). Replaces the 
+    range indicated by the colon with all residues in that range and 
+    keeps all residues separated by commas. Removes duplicates.
+    """
+
+    res_list = []
+    # Split by comma and then colon to make list of list
+    split_list = [elem.split(':') for elem in user_input.split(',')]
+    for sub_list in split_list:
+        # If list size is one, word is separated by comma
+        if len(sub_list) == 1:
+            try:
+                res = sub_list[0]
+                # Find index
+                identifiers.index(res)
+                # Append to list
+                res_list.append(res)
+            except:
+                raise ValueError(f"Residue not in PDB: {res}")
+        # If list size is 2, word is separated by a single colon
+        elif len(sub_list) == 2:
+            try:
+                # Get indexes for residue in identifiers
+                u_idx, v_idx = [identifiers.index(res) for res in sub_list]
+            except:
+                raise ValueError(f"Residue range not in PDB: {':'.join(sub_list)}")
+            # Check if order of residues is reversed
+            if u_idx >= v_idx:
+                raise ValueError(f"Range not specified correctly: {':'.join(sub_list)}")
+            try:
+                # This block should not cause an error
+                # Create list with all residues in that range
+                res_range = identifiers[u_idx:v_idx + 1]
+                # Concatenate with output list
+                res_list += res_range
+            except Exception as e:
+                raise e
+        # Other list sizes means multiple colons
+        else:
+            err_str = f"Incorrect format, only one ':' allowed: {':'.join(sub_list)}"
+            raise ValueError(err_str)
+    # Remove duplicates
+    res_list = list(set(res_list))
+    return res_list
+
+def get_hubs(G, **kwargs):
```

Comment:
`**kwargs` can lead to unexpected behaviours because you have no control over them. It is better to explicit each argument passed to the function. For those arguments that need to be passed directly to a `networkx` function, use the same name they have in the `networkx` function. Check and modify all the other functions accepting `**kwargs`.

Reply:
I'm not sure I understand what kind of unexpected behaviors can be caused by `**kwargs` since I only call these functions with a specific dictionary of arguments and I manually choose which arguments are in that dictionary. 
The idea behind using `**kwargs` was to make the `get_centrality_dict` function future proof. So if we ever need to add any new centrality measures (networkX or custom) then we can simply write the function separately, and just add its name to the function map and it will work regardless of what arguments it might require. Without it, the `get_centrality_dict`  will get longer with every new centrality measure since we have to perform a check to see the name of the measure and then call it from the function map using the arguments specific for that measure. 
Is the issue that if a function has `**kwargs` it can accept multiple arguments? If that is the case would it be okay if I just added a single argument for an argument dictionary, and then depending on the function it would only choose the arguments it requires from that dictionary?

Reply:
I changed the names of the dictionary entries to match the networkX argument names

Reply:
It's not really the `**kwargs` in `get_centrality dict` but in all the other functions implementing centrality measures, since leaving the `**kwargs` in them would mean having no control over what is passed to the internal calls. Maybe let's hear also @mtiberti's opinion on this :)